plugins/sdm/accessbot.py

```diff
@@ -87,6 +92,7 @@ def access_resource(self, message, match):
             return
         resource_name = self.get_arguments_helper().remove_flags(arguments)
         flags = self.get_arguments_helper().extract_flags(arguments, validators=self.get_resource_grant_helper().get_flags_validators())
+        self.check_requester_flag(message, flags.get('requester'))
```

Comment:
Why not doing this in a validator... ?

Reply:
1. The validation methods check flag values. For example: the `reason_flag_validator` method checks that the value of the "reason" flag is not empty; the `duration_flag_validator` method checks whether the value of the "duration" flag has a valid format (number + letter);
2. We do not validate the value of the "requester" flag because it is not the user who defines this value, as this flag plus its value are automatically placed in the command by the Workflow Builder;
3. We only check if the flag was used and if the user who used it was a bot (that's why we call the function "check", not "validator"). If a user tries to use this flag, a message will be returned on the channel informing him that he cannot do this.
4. Talking with Daniel, we came to the conclusion that it would be inappropriate to add these two checks to a validator.